File: engine/engine-model/src/main/scala/net/server/channel/packet/wedding/AddRegistryItemPacket.scala
package net.server.channel.packet.wedding

class AddRegistryItemPacket(private var _mode: Byte, private var _slot: Short, private var _itemId: Int, private var _quantity: Short) extends BaseWeddingPacket(_mode = _mode) {
  def slot: Short = _slot

  def itemId: Int = _itemId

  def quantity: Short = _quantity
}

File: engine/engine-model/src/main/java/net/server/channel/packet/wedding/AddRegistryItemPacket.java
package net.server.channel.packet.wedding;

public class AddRegistryItemPacket extends BaseWeddingPacket {
   private final Short slot;

   private final Integer itemId;

   private final Short quantity;

   public AddRegistryItemPacket(Byte mode, Short slot, Integer itemId, Short quantity) {
      super(mode);
      this.slot = slot;
      this.itemId = itemId;
      this.quantity = quantity;
   }

   public Short slot() {
      return slot;
   }

   public Integer itemId() {
      return itemId;
   }

   public Short quantity() {
      return quantity;
   }
}
